aws/data_source_aws_lambda_layer_version.go

```diff
@@ -0,0 +1,150 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func dataSourceAwsLambdaLayerVersion() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsLambdaLayerVersionRead,
+
+		Schema: map[string]*schema.Schema{
+			"layer_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"version": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"compatible_runtimes"},
+			},
+			"compatible_runtime": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ValidateFunc:  validation.StringInSlice(validLambdaRuntimes, false),
+				ConflictsWith: []string{"version"},
+			},
+			"compatible_runtimes": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				MaxItems: 5,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"license_info": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"layer_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"source_code_hash": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"source_code_size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsLambdaLayerVersionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lambdaconn
+	layerName := d.Get("layer_name").(string)
+
+	var version int64
+
+	if v, ok := d.GetOk("version"); ok {
+		version = int64(v.(int))
+	} else {
+		listInput := &lambda.ListLayerVersionsInput{
+			LayerName: aws.String(layerName),
+		}
+		if v, ok := d.GetOk("compatible_runtime"); ok {
+			listInput.CompatibleRuntime = aws.String(v.(string))
+		}
+
+		log.Printf("[DEBUG] Looking up latest version for lambda layer %s", layerName)
+		listOutput, err := conn.ListLayerVersions(listInput)
```

Comment:
To prevent potential problems in the future, this API call should likely be updated to use the [`Marker`](https://docs.aws.amazon.com/sdk-for-go/api/service/lambda/#ListLayerVersionsInput) parameter for paginated results since the SDK does not provide a paginated function. 👍  I don't want to hold up this pull request on this for an initial release though.

Reply:
something I noticed is that the ListLayerVersions api returns results in descending order, and we only ever get the first one - would it still need pagination in that case?